src/main/java/edu/cornell/kfs/vnd/document/validation/impl/CuVendorRuleBase.java

```diff
@@ -57,8 +57,8 @@ protected boolean validateParentVendorTaxNumber(VendorDetail vendorDetail) {
         // negative criteria so that the current vendor is excluded from the search
         if (ObjectUtils.isNotNull(vendorDetail.getVendorHeaderGeneratedIdentifier())) {
             // ==== CU Customization: Use CriteriaLookupService instead, since BO service doesn't allow negative criteria in non-count methods. ====
-            criteria.add(PredicateFactory.notEqual(VendorPropertyConstants.VENDOR_HEADER_GENERATED_ID,
-                    vendorDetail.getVendorHeaderGeneratedIdentifier()));
+            criteria.add(PredicateFactory.notEqualIgnoreCase(VendorPropertyConstants.VENDOR_HEADER_GENERATED_ID,

```

Comment:
updated

Reply:
If you don't mind, for my own enlightenment and understanding...  I don't understand why notEqualIgnoreCase() method would not work for numbers, whereas notEqual() would. Can you explain that?

Reply:
@dp65,

The vendor's header ID field is stored as a number in the database, not as a string containing numeric digits. The `notEqualIgnoreCase()` check is set up so that it will ultimately compare the column's value against a string value. Such a comparison may theoretically still work due to implicit type conversion, but it would be considered bad practice because the data types don't line up. That's why a `notEqual()` comparison between similar data types would be preferable.

Reply:
@cah292 , Ok, I followed that.  Does that imply that notEqual() is not supposed to be used to compare string values? Or, alternatively, is it designed to accept either numbers or strings?  And If it is, why didn't KualiCo, design notEqualIgnoreCase to accept both types also. Especially if their intent was to replace notEqual() usage with notEqualIgnoreCase() ?

Reply:
@dp65 The `notEqual()` comparison is set up to support multiple data types (string, number, date, etc.). The `notEqualIgnoreCase()` comparison only accepts a string or other textual value to compare against; also, its name already implies that the data being compared is a string (or can be converted to a string), since an "ignore case" aspect suggests that the data is textual in nature.

As for why the `notEqual()` method was removed in the first place, that was likely an accident. KualiCo removed a fair amount of unused KFS code for the 2022-05-25 upgrade (or, at least code that wasn't actively used by a non-customized system). However, KualiCo must not have considered the implications of removing the `notEqual()` method from a factory class, given that that class was supposed to be flexible enough to support most types of query criteria. Perhaps KualiCo used some kind of tool to detect "dead" code but didn't fully follow up on whether the removal of each item actually made sense.

Reply:
@cah292 , Thank you SO much for the extra details about KualiCo's behavior.  That makes a lot more sense to me now.